backend/middleware/validation.js

```
@@ -1,53 +1,120 @@
+    _logger.log(msg, meta);
+  }
+}
+
+// Recursively sanitize all string fields in an object, respecting schema if provided
+function deepSanitize(obj, schema = {}, path = []) {
+  if (!obj || typeof obj !== 'object') return obj;
+  for (const key of Object.keys(obj)) {
+    const value = obj[key];
+    const fieldPath = [...path, key].join('.');
+    const fieldRule = schema[key] || { sanitize: true, mode: 'plain' };
+    if (fieldRule && fieldRule.sanitize) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)